daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Decimal.scala

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package data
+
+import java.math.BigDecimal
+
+import scala.math.{BigDecimal => BigDec}
+
+// Our legacy Numerics are fix scale 10, aka Decimals
+// This object provides some legacy utility functions for Decimal
+@deprecated("Use Numeric instead", since = "3.0.0")
+object Decimal {
+
+  @deprecated("Use Numeric.Scale.assertFromInt(10)", since = "3.0.0")
+  val scale: Numeric.Scale = Numeric.Scale.assertFromInt(10)
+
+  @deprecated("Use Numeric.Scale.assertFromInt(10).MaxValue", since = "3.0.0")
+  def MaxValue: Numeric = Numeric.maxValue(scale)
+  @deprecated("Use Numeric.Scale.assertFromInt(10).MinValue", since = "3.0.0")
+  def MinValue: Numeric = Numeric.minValue(scale)
+
+  @deprecated("Use Numeric.fromBigDecimal(Scale.assertFromInt(10), _)", since = "3.0.0")
+  def fromBigDecimal(x: BigDec): Either[String, data.Numeric.Numeric] =
+    Numeric.fromBigDecimal(scale, x)
+
+  private val hasExpectedFormat =
+    """[+-]?\d{1,28}(\.\d{1,10})?""".r.pattern
+
+  @deprecated(
+    """Use Numeric.fromString(Scale.assertFromInt(10), _). Note this alternative is stricter as it requires the string to match "`-?[1-9]\d{1,28}.([1-9]{10})`"""",
```

Comment:
@remyhaemmerle-da : what is the reason for the more strict matching?

I've just run into that causing a test-failure.

Reply:
That said. We've decided to become more strict in our decoding as well. You can ignore this comment.

Reply:
The reason is that I do not want to write and maintain a new parser. 
In `Numeric.fromString(_, _)` reuses `Numeric.fromString(_)` that uses the number of digits on the right and side of the '.' to determine the scale.

If you want something more lenient, I suggest you use BigDecimal parser and convert the result with `Numeric.fromBigDecimal` 

```
def convert(s: String) =  Numeric.assertFromBigDecimal(Numeric.Scale.assertFromInt(10), BigDecimal(s))
```

Reply:
ack. thx. I ended up going that path while keeping the format restriction we've had for Decimal.